Not able to start K3s in CI builds

### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

```
Run curl -sfL https://get.k3s.io/ | INSTALL_K3S_VERSION= INSTALL_K3S_CHANNEL=stable INSTALL_K3S_EXEC=--docker K3S_KUBECONFIG_MODE=644 sh -
[INFO]  Finding release for channel stable
[INFO]  Using stable as release
[INFO]  Downloading hash https://github.com/k3s-io/k3s/releases/download/stable/sha256sum-amd64.txt
Error: Process completed with exit code 22.
```

### Version

latest

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
.
```


### Logs from the workflow controller

```text
kubectl logs -n argo deploy/workflow-controller | grep ${workflow}
```


### Logs from in your workflow's wait container

```text
kubectl logs -n argo -c wait -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded
```

--- a/.github/workflows/ci-build.yaml
+++ b/.github/workflows/ci-build.yaml
@@ -68,24 +68,34 @@ jobs:
         include:
           - test: test-executor
             profile: minimal
+            install_k3s_version: v1.26.3+k3s1
           - test: test-corefunctional
             profile: minimal
+            install_k3s_version: v1.26.3+k3s1
           - test: test-functional
             profile: minimal
+            install_k3s_version: v1.26.3+k3s1
           - test: test-api
             profile: mysql
+            install_k3s_version: v1.26.3+k3s1
           - test: test-cli
             profile: mysql
+            install_k3s_version: v1.26.3+k3s1
           - test: test-cron
             profile: minimal
+            install_k3s_version: v1.26.3+k3s1
           - test: test-examples
             profile: minimal
+            install_k3s_version: v1.26.3+k3s1
           - test: test-plugins
             profile: plugins
+            install_k3s_version: v1.26.3+k3s1
           - test: test-java-sdk
             profile: minimal
+            install_k3s_version: v1.26.3+k3s1
           - test: test-python-sdk
             profile: minimal
+            install_k3s_version: v1.26.3+k3s1
           - test: test-executor
             install_k3s_version: v1.21.2+k3s1
             profile: minimal

--- a/.github/workflows/ci-build.yaml
+++ b/.github/workflows/ci-build.yaml
@@ -118,9 +118,9 @@ jobs:
       - name: Install and start K3S
         run: |
           if ! echo "${{ matrix.install_k3s_version }}" | egrep '^v[0-9]+\.[0-9]+\.[0-9]+\+k3s1$'; then
-            INSTALL_K3S_VERSION=v1.26.3+k3s1
+            export INSTALL_K3S_VERSION=v1.26.3+k3s1
           else
-            INSTALL_K3S_VERSION=${{ matrix.install_k3s_version }}
+            export INSTALL_K3S_VERSION=${{ matrix.install_k3s_version }}
           fi
           curl -sfL https://get.k3s.io | INSTALL_K3S_CHANNEL=stable INSTALL_K3S_EXEC=--docker K3S_KUBECONFIG_MODE=644 sh -
           until kubectl --kubeconfig=/etc/rancher/k3s/k3s.yaml cluster-info ; do sleep 10s ; done

--- a/.github/workflows/ci-build.yaml
+++ b/.github/workflows/ci-build.yaml
@@ -68,34 +68,24 @@ jobs:
         include:
           - test: test-executor
             profile: minimal
-            install_k3s_version: v1.26.3+k3s1
           - test: test-corefunctional
             profile: minimal
-            install_k3s_version: v1.26.3+k3s1
           - test: test-functional
             profile: minimal
-            install_k3s_version: v1.26.3+k3s1
           - test: test-api
             profile: mysql
-            install_k3s_version: v1.26.3+k3s1
           - test: test-cli
             profile: mysql
-            install_k3s_version: v1.26.3+k3s1
           - test: test-cron
             profile: minimal
-            install_k3s_version: v1.26.3+k3s1
           - test: test-examples
             profile: minimal
-            install_k3s_version: v1.26.3+k3s1
           - test: test-plugins
             profile: plugins
-            install_k3s_version: v1.26.3+k3s1
           - test: test-java-sdk
             profile: minimal
-            install_k3s_version: v1.26.3+k3s1
           - test: test-python-sdk
             profile: minimal
-            install_k3s_version: v1.26.3+k3s1
           - test: test-executor
             install_k3s_version: v1.21.2+k3s1
             profile: minimal
@@ -127,7 +117,12 @@ jobs:
           cache: pip
       - name: Install and start K3S
         run: |
-          curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=${{matrix.install_k3s_version}} INSTALL_K3S_CHANNEL=stable INSTALL_K3S_EXEC=--docker K3S_KUBECONFIG_MODE=644 sh -
+          if ! echo "${{ matrix.install_k3s_version }}" | egrep '^v[0-9]+\.[0-9]+\.[0-9]+\+k3s1$'; then
+            INSTALL_K3S_VERSION=v1.26.3+k3s1
+          else
+            INSTALL_K3S_VERSION=${{ matrix.install_k3s_version }}
+          fi
+          curl -sfL https://get.k3s.io | INSTALL_K3S_CHANNEL=stable INSTALL_K3S_EXEC=--docker K3S_KUBECONFIG_MODE=644 sh -
           until kubectl --kubeconfig=/etc/rancher/k3s/k3s.yaml cluster-info ; do sleep 10s ; done
           cp /etc/rancher/k3s/k3s.yaml /home/runner/.kubeconfig
           echo "- name: fake_token_user" >> $KUBECONFIG

